### ISSUE TITLE
Endian problem in SOCKS support

### ISSUE THREAD

**Issue by GoogleCodeExporter** [transmission-remote-gui/transgui#672]
```
What steps will reproduce the problem?
1. Set up a socks proxy (e.g. ssh’s -D dynamic proxy)
2. Attempt to connect to a transmission daemon on the remote end using the 
normal port of 9091
3. Note that transmission-remote-gui attempts to connect to port 33571 instead.

9091 in hexadecimal is 0x2383
33571 in hexadecimal is 0x8323.

So the bytes of the port are being swapped around (or more likely *not* being 
swapped when they should be: x86 architecture is little-endian while network 
order is big-endian).
```

Original issue reported on code.google.com by `ha...@hawkesnest.net` on 5 Jul 2013 at 6:27

**Comment by choueric** [comment]
I am writting a socks5 proxy to connect transmission running on my NAS by trans-remote-gui, which runs on ubuntu(x64). The connect request sent by remote-gui is [5 1 0 1 127 0 0 1 131 35], which shows host is 127.0.0.1 and port number is 0x8323(33571) because the rfc says port number is "desired destination port in network octet order".

BTW, the proxy works fine if it treats the port number in little endian way when the value is 33571 and big endian otherwise. HTTP and ssh run well via it, too.

**Comment by antekgla** [comment]
Are you running TransGUI in linux?

**Comment by Grief** [comment]
@antekgla Yes, I am

**Comment by PeterDaveHello** [comment]
@antekgla We're using Debian Stretch

**Comment by antekgla** [comment]
In which platform? 
I try to find if the platform is Big Endian or Little Endian...

**Comment by PeterDaveHello** [comment]
@antekgla depends on the ones you download for, amd64, i386, armv6, armv7.

**Comment by antekgla** [comment]
@Grief what is your version of TransGUI (amd64, i386, armv6, armv7) ?

**Comment by Grief** [comment]
@antekgla amd64

**Comment by antekgla** [comment]
Well so I discard what your platform is big endian.

Meanwhile...do you try changing the Transmission port setting in TransGUI to 33571 (if your transmission daemon port is 9091)?

**Comment by Grief** [comment]
@antekgla right, sorry, that I didn't answer you first. Yes, the workaround with 'endian-mirroring' of the port is working.
